src/test/java/io/jenkins/plugins/analysis/warnings/recorder/TrendCarouselITest.java

```diff
@@ -0,0 +1,271 @@
+        carousel.next();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+    }
+
+    /**
+     * Test that severities, new versus fixed and tools trend charts are shown in this order by clicking previous.
+     */
+    @Test
+    public void shouldShowTrendChartsInRightOrderByPrevious() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
+        carousel.previous();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+        carousel.previous();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.TOOLS);
+    }
+
+    /**
+     * Test that severities trend chart is shown after clicking next and previous and clicking previous and next.
+     */
+    @Test
+    public void shouldShowSeveritiesTrendChartAfterNextAndPrevious() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
+        carousel.next();
+        carousel.previous();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
+        carousel.previous();
+        carousel.next();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.SEVERITIES);
+    }
+
+    /**
+     * Test that new versus fixed trend chart is shown after clicking previous and also after refreshing the page
+     * (loaded from local storage).
+     */
+    @Test
+    public void shouldShowNewVersusFixedTrendChartAfterPreviousAndReload() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        carousel.previous();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+        try {
+            webPage.refresh();
+        }
+        catch (IOException e) {
+            throw new AssertionError("WebPage refresh failed.", e);
+        }
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.NEW_VERSUS_FIXED);
+    }
+
+    /**
+     * Test that tools trend chart is shown after clicking next and also after refreshing the page (loaded from local
+     * storage).
+     */
+    @Test
+    public void shouldShowToolsTrendChartAfterNextAndReload() {
+        TrendCarousel carousel = setUpTrendChartTest(false);
+
+        carousel.next();
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.TOOLS);
+        try {
+            webPage.refresh();
+        }
+        catch (IOException e) {
+            throw new AssertionError("WebPage refresh failed.", e);
+        }
+        assertThat(carousel.getActiveChartType()).isEqualTo(TrendChartType.TOOLS);
+    }
+
+    /**
+     * Tests that only one of all three charts is shown as active.
+     */
+    @Test
+    public void shouldShowOnlyOneOfThreeChartsAsActive() {
```

Comment:
This is the same test as `shouldShowSeveritiesTrendChartAsDefault`just with more assertions.